### ISSUE TITLE
FixedChannelPool pool.acquire().syncUninterruptibly().get()  blocked for ever

### ISSUE THREAD

**Issue by ppluck** [netty/netty#10224]
### Expected behavior
       expect can get the connection
### Actual behavior
      wait for ever.  blocked.  The problem happened in  DefaultPromise.awaitUninterruptibly(), Line number 274  (wait());  The same problem happened in DefaultPromise.await().
### Steps to reproduce
    env : http://127.0.0.1/test.txt  can access 
    FixedChannelPool  or  SimpleChannelPool has the same problem.
    no pool is ok.

### Minimal yet complete reproducer code (or URL to code)
The demo code can be executed by main method.

    ` static final String URL = "http://127.0.0.1/test.txt";

    public static void main(String[] args) throws Exception {
        URI uri = new URI(URL);
        String scheme = uri.getScheme() == null? "http" : uri.getScheme();
        String host = uri.getHost() == null? "127.0.0.1" : uri.getHost();
        int port = uri.getPort();
        if (port == -1) {
            if ("http".equalsIgnoreCase(scheme)) {
                port = 80;
            } else if ("https".equalsIgnoreCase(scheme)) {
                port = 443;
            }
        }

        if (!"http".equalsIgnoreCase(scheme) && !"https".equalsIgnoreCase(scheme)) {
            System.err.println("Only HTTP(S) is supported.");
            return;
        }

        // Configure SSL context if necessary.
        final boolean ssl = "https".equalsIgnoreCase(scheme);
        final SslContext sslCtx;
        if (ssl) {
            sslCtx = SslContextBuilder.forClient()
                .trustManager(InsecureTrustManagerFactory.INSTANCE).build();
        } else {
            sslCtx = null;
        }

        // Configure the client.
        try {
            Bootstrap b = new Bootstrap();
            b.group(group)
             .channel(NioSocketChannel.class)
             .handler(new HttpSnoopClientInitializer(sslCtx));

            SchemeHostPort shp = new SchemeHostPort("http", "10.16.164.113", 80);
            SocketAddress remoteAddress = new InetSocketAddress(shp.getHost(), shp.getPort());
    		b.remoteAddress(remoteAddress);
    		
    		
    		FixedChannelPool pool = new FixedChannelPool(b, new HttpChannelPoolHandler(sslCtx), ChannelHealthChecker.ACTIVE,
    				AcquireTimeoutAction.NEW, acquireTimeoutMillis, maxConnections, maxPendingAcquires, releaseHealthCheck,
    				lastRecentUsed);
    		
    		
            // Make the connection attempt.
     //    		Channel ch = b.connect(host, port).sync().channel();
     //            Channel ch = b.connect().sync().channel();
     //            Channel ch = pool.acquire().syncUninterruptibly().get();

            Future<Channel> fc = pool.acquire();
            fc = fc.syncUninterruptibly();  //wait for ever. Instance is DefaultPromise, Method is awaitUninterruptibly(), Line number 274  (wait());   blocked for ever
            Channel ch = fc.get();
            
            
            // Prepare the HTTP request.
            HttpRequest request = new DefaultFullHttpRequest(
                    HttpVersion.HTTP_1_1, HttpMethod.GET, uri.getRawPath(), Unpooled.EMPTY_BUFFER);
            request.headers().set(HttpHeaderNames.HOST, host);
            request.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.KEEP_ALIVE);
            request.headers().set(HttpHeaderNames.ACCEPT_ENCODING, HttpHeaderValues.GZIP);

            // Set some example cookies.
            request.headers().set(
                    HttpHeaderNames.COOKIE,
                    ClientCookieEncoder.STRICT.encode(
                            new DefaultCookie("my-cookie", "foo"),
                            new DefaultCookie("another-cookie", "bar")));

            // Send the HTTP request.
            ch.writeAndFlush(request);

            // Wait for the server to close the connection.
            ch.closeFuture().sync();
        } finally {
            // Shut down executor threads to exit.
            group.shutdownGracefully();
        }
    }`

### Netty version
	<groupId>io.netty</groupId>
	<artifactId>netty-all</artifactId>
	<version>4.1.49.Final</version>   
        
        Problem version is 4.1.49.Final.
        Success version is 4.1.22.Final.

### JVM version (e.g. `java -version`)
    java version "1.8.0_241"
    Java(TM) SE Runtime Environment (build 1.8.0_241-b07)
    Java HotSpot(TM) 64-Bit Server VM (build 25.241-b07, mixed mode)

### OS version (e.g. `uname -a`)
    win10

**Comment by ppluck** [comment]
Who can help me? Thank you.

**Comment by ppluck** [comment]
This problem has been solved.  
The cause is : 
netty 4.1.49  : When the ChannelPoolHandler(Implementation class).channelcreated method is called, the ChannelPoolHandler(Implementation class).channelacquired method will be called.
netty  :  When the ChannelPoolHandler(Implementation class).channelcreated method is called, the ChannelPoolHandler(Implementation class).channelacquired method will not be called.

I called the same logical method in both methods.  
Solution: remove the call to this method from ChannelPoolHandler(Implementation class).channelcreated
